### PR TITLE
Add timestamp to events

### PR DESCRIPTION
**Pull request by windmemory**
Refer to issue: https://github.com/Chatie/wechaty/issues/1829

### CODE REVIEW

File: src/puppet.ts [hunk 1]
```diff
@@ -289,21 +289,21 @@ export abstract class Puppet extends EventEmitter {
...
+  public on (event: 'error',        listener: (error: string) => void)                                                                                     : this
+  public on (event: 'friendship',   listener: (friendshipId: string) => void)                                                                              : this
+  public on (event: 'login',        listener: (contactId: string) => void)                                                                                 : this
+  public on (event: 'logout',       listener: (contactId: string) => void)                                                                                 : this
+  public on (event: 'message',      listener: (messageId: string) => void)                                                                                 : this
+  public on (event: 'reset',        listener: (reason: string) => void)                                                                                    : this
+  public on (event: 'room-join',    listener: (roomId: string, timestamp: number, inviteeIdList: string[], inviterId:  string) => void)                    : this
```
Comment (huan) [inline src/puppet.ts]: The same:

1. move `timestamp` to the end of the argument list
1. remove unrelated changes. (align)

**Comment by windmemory** [comment]
Modified code according to review.

File: src/puppet.ts [hunk 2]
```diff
@@ -257,19 +257,19 @@ export abstract class Puppet extends EventEmitter {
...
-  public emit (event: 'reset',        reason: string)                                                                                   : boolean
-  public emit (event: 'room-join',    roomId: string, timestamp: number, inviteeIdList:  string[], inviterId: string)                   : boolean
-  public emit (event: 'room-leave',   roomId: string, timestamp: number, leaverIdList:   string[], remover?: string)                    : boolean
-  public emit (event: 'room-topic',   roomId: string, timestamp: number, newTopic:       string,   oldTopic: string, changerId: string) : boolean
-  public emit (event: 'room-invite',  roomInvitationId: string)                                                                         : boolean
-  public emit (event: 'scan',         qrcode: string, status: ScanStatus, data?: string)                                                : boolean
-  public emit (event: 'ready')                                                                                                          : boolean
+  public emit (event: 'dong',         data?: string)                                                                 : boolean
```
Comment (huan) [inline src/puppet.ts]: Too many unrelated changes. Please revert them.

File: src/puppet.ts [hunk 3]
```diff
@@ -257,19 +257,19 @@ export abstract class Puppet extends EventEmitter {
...
+  public emit (event: 'error',        error: Error)                                                                  : boolean
+  public emit (event: 'friendship',   friendshipId: string)                                                          : boolean
+  public emit (event: 'login',        contactId: string)                                                             : boolean
+  public emit (event: 'logout',       contactId: string)                                                             : boolean
+  public emit (event: 'message',      messageId: string)                                                             : boolean
+  public emit (event: 'reset',        reason: string)                                                                : boolean
+  public emit (event: 'room-join',    roomId: string, inviteeIdList:  string[], inviterId: string)                   : boolean
```
Comment (huan) [inline src/puppet.ts]: How can you get to know the timestamp for the listeners, when you do not have a timestamp argument for emit?

**Comment by huan** [comment]
Please bump the patch number for this module, so that the code could be able to be published after this PR had been merged.

**Comment by windmemory** [comment]
Version bumped.

**Comment by huan** [comment]
Thanks for enhance the puppet system!
